website/content/docs/enterprise/ent-to-ce-downgrades.mdx

```diff
@@ -0,0 +1,110 @@
+  Guide to downgrade from consul enterprise to consul ce
+---
+
+## Introduction
+
+This document describes some best practices that you should follow when
+downgrading from Consul enterprise to consul ce.
+
+## Download the CE Version
+
+First, download the binary for the CE version you want.
+
+<Tabs>
+<Tab heading="Binary">
+
+All current and past versions of the CE and Enterprise releases are
+available here:
+
+- https://releases.hashicorp.com/consul
+
+</Tab>
+<Tab heading="Docker">
+
+Docker containers are available at this location:
+
+- **CE:** https://hub.docker.com/_/consul
+
+</Tab>
+
+</Tabs>
+
+## Prepare for the Downgrade to CE
+
+**1.** Take a snapshot:
+
+```
+consul snapshot save backup.snap
+```
+
+You can inspect the snapshot to ensure if was successful with:
+
+```
+consul snapshot inspect backup.snap
+```
+
+Example output:
+
+```
+ID           2-1182-1542056499724
+Size         4115
+Index        1182
+Term         2
+Version      1
+```
+
+This will ensure you have a safe fallback option in case something goes wrong. Store
+this snapshot somewhere safe. More documentation on snapshot usage is available here:
+
+- [consul.io/commands/snapshot](/consul/commands/snapshot)
+- [Backup Consul Data and State tutorial](/consul/tutorials/production-deploy/backup-and-restore)
+
+**2.** Temporarily modify your Consul configuration so that its [log_level](/consul/docs/agent/config/cli-flags#_log_level)
+is set to `debug`. After doing this, issue the following command on your servers to
+reload the configuration:
+
+```
+consul reload
+```
+
+This change will give you more information to work with in the event something goes wrong.
+
+
+**3.** Remove references to tenanted services from service-resolver, service-splitter and service-router config entries.
+
+
+## Perform the Downgrade
+
+- Restart or redeploy all clients with a CE version of the binary
+
+- Update server binaries to use the CE version
+
+- Restart or redeploy the servers one at a time with an env var named `CONSUL_ENTERPRISE_DOWNGRADE_TO_CE` set to `true` on the servers.
+
+- Validate that all servers are on CE and healthy using `consul members` command
+
+## What happens during the downgrade
+
+During the downgrade as the raft replication logs start flowing into the ce server we handle them in one of the following ways :- 
+
+- Drop the request
+
+   For example :-
+   - Register requests - register requests having non default namespace, having services or health checks in non default namespace or partition are dropped.
+
+   - Peering Write Requests - peering write requests are dropped if the local partiton connecting to the peer is non default. 
+
+- Filter out tenanted data from the request
+
+  For example :-
+  - Acl Token Batch Set requests -  acl  tokens with non default namespace or partiton are filtered out.
+  - Acl Policy Batch Set requests -  acl  policies with non default namespace or partiton are filtered out.
+
+- Panic and stop the downgrade
```

Comment:
It is probably worth an extra sentence here to note why we have to panic. Something to the effect of:

> Config Entries regarding traffic routing cannot be safely filtered because Consul  cannot know whether the resulting filtered config entries send traffic towards  services which can handle that traffic. In these situations, you are required to first remove references to services within non default namespaces or partitions from within those config entries. When encountering these config entries Consul CE in the downgrade mode will panic in order to not cause harm to existing service mesh routing.